rust/agama-lib/src/store.rs

```diff
@@ -94,10 +94,10 @@ impl Store {
     pub async fn store(&self, settings: &InstallSettings) -> Result<(), ServiceError> {
         if let Some(scripts) = &settings.scripts {
             self.scripts.store(scripts).await?;
-        }
 
-        if settings.scripts.as_ref().is_some_and(|s| !s.pre.is_empty()) {
-            self.run_pre_scripts().await?;
+            if scripts.pre.as_ref().is_some_and(|s| !s.is_empty()) {
```

Comment:
here it is kind of strange...can it happen at all to be some and also empty? From code it looks to me like it cannot happen together.

Reply:
I would say that, in JSON, it can be "pre: []", so it is `Some` but empty.